### DISCUSSION TITLE
Which extension is compatible with gauge project in azure devops to publish html-report

### DISCUSSION THREAD

**Discussion by fundabdeniz** [getgauge/gauge#2279] [General]
### Actual behavior
I used HtML Viewer to publish report in azure pipeline.Artifact is produced after pipeline is passed. I could download to my local and run open index.html. Could I publish html report below HTML-Viwever tab such as dashboard to publish html report via extension in azure devops. Could you suggest any other extension compatible with gauge format(written jvascript)?

### Gauge version
```
Gauge version: 1.4.3 
flash (0.0.2) html-report (4.0.10) java (0.9.1) js (2.4.0) screenshot (0.1.0)
```

**Comment by zabil** [comment]
You can use the xml report plugin to generate Junit compatible test report. Tools usually parse them to display the output.

**Comment by fundabdeniz** [comment]
I used javascript while otomate test case. So is it compatible with javascript based.  Artifact includes files like scrrenshot.  Extension should be html suitability.
![image](https://user-images.githubusercontent.com/75672044/185552557-4df18a16-ef88-4d0b-bd67-b90b86b23584.png)

this  task didn't report index.html  
![image](https://user-images.githubusercontent.com/75672044/185553171-63019d75-c156-4ef6-ae44-d4809a0f35cb.png)

**Comment by zabil** [comment]
Please use the xml-report plugin.

```
gauge install xml-report
```

Change the "Search Folder" to
```
${Build.Repository.LocalPath)\gauge\web\walvent-admin\xml-report
```

This should show you the reports.

**Reply by fundabdeniz** [reply]
Hi,

I have an issue using this plugin . (https://github.com/getgauge/gauge/issues/2281)
I could not report all test specs. 

Could you suggest anything?

**Comment by fundabdeniz** [comment]
Thanks for your feedback.  Report is published successfull  after applying your knowledge.

**Comment by fundabdeniz** [comment]
After install plugin xml-report, I could publish dashboard like below **screenshot. Artifact has include result.xml .Report file has not specs folder becasue of that chart has no all specs or test case.  @zabil   Could I have all specs in the xml chart 

Pipeline tasks:
![Pipeline-tasks](https://user-images.githubusercontent.com/75672044/185901240-39df1a2c-fba3-4a18-a843-0f32271b2927.png)

![Artfiact](https://user-images.githubusercontent.com/75672044/185900361-891c1ee5-add6-48db-8926-08f87acfa17f.png)

![xml-report chart](https://user-images.githubusercontent.com/75672044/185899836-b55ca087-9b84-4184-a108-7a50674c7c47.png)
